docs/DeveloperGuide.md

```diff
@@ -1,33 +1,297 @@
+
+* [`UI`](https://github.com/AY2324S1-CS2113-T17-1/tp/tree/master/src/main/java/athleticli/ui): The UI and other UI-related sub-components of AthletiCLI.
+* [`Parser`](https://github.com/AY2324S1-CS2113-T17-1/tp/tree/master/src/main/java/athleticli/parser): Parses the commands input by the users.
+* [`Storage`](https://github.com/AY2324S1-CS2113-T17-1/tp/tree/master/src/main/java/athleticli/storage): Reads data from, and writes data to, the hard disk.
+* [`Data`](https://github.com/AY2324S1-CS2113-T17-1/tp/tree/master/src/main/java/athleticli/data): Holds the data of AthletiCLI in memory.
+* [`Commands`](https://github.com/AY2324S1-CS2113-T17-1/tp/tree/master/src/main/java/athleticli/commands): The command executors.
+
+[`Exceptions`](https://github.com/AY2324S1-CS2113-T17-1/tp/tree/master/src/main/java/athleticli/exceptions) represents exceptions used by multiple other components.
+
+### Overview
+
+The class diagram shows the relationship between `AthletiCLI`, `Ui`, `Parser`, and `Data`.
+
+<p  align="center" width="100%">
+  <img width="80%" src="images/MainClassDiagram.svg" alt="'set-diet-goal' Sequence Diagram"/>
+</p>
+
+### Data Component
+
+The class diagram shows how the `Data` component is constructed with multiple classes.
+
+<p  align="center" width="100%">
+  <img width="80%" src="images/DataClassDiagram.svg" alt="'set-diet-goal' Sequence Diagram"/>
+</p>
+
+**How the architecture components interact with each other**
+
+The _Sequence Diagram_ below shows how the components interact with each other for the scenario where the user issues the command `help add-diet`.
+
+![](images/HelpAddDiet.svg)
+
+This diagram involves the interaction between `AthletiCLI`, `UI`, `Parser`, `Commands` components and the user.
+
+The `Storage` component only interacts with the `Data` component. The _Sequence Diagram_ below shows how they interact with each other for the scenario where a `save` command is executed.
+
+![](images/Save.svg)
+
+For simplicity, only 1 `StorableList` is drawn instead of the actual 6.
+
+## Implementation
+
+### Diet Management in AthletiCLI
+
+#### [Implemented] Setting Up, Editing, Deleting, Listing, and Finding Diets
+
+Regardless of the operation you are performing on diets (setting up, editing, deleting, listing, or finding), the process follows a general five-step pattern in AthletiCLI:
+
+1. **Input Processing**: The user's input is passed through AthletiCLI to the Parser Class. Examples of user inputs include:
+    - `add-diet calories/500 protein/20 carb/50 fat/10 datetime/2021-09-01 06:00` for adding a diet.
+    - `edit-diet 1 calories/500 protein/20 carb/50 fat/10 datetime/2021-09-01 06:00` for editing a diet.
+    - `delete-diet 1` for deleting a diet.
+    - `list-diet` for listing all diets.
+    - `find-diet 2021-09-01` for finding diets of a particular date.
+
+2. **Command Identification**: The Parser Class identifies the type of diet operation and passes the necessary parameters.
```

Comment:
In published website, the numbering reads 1., 7, ..., 10. Is there something that can be done about this?